usaspending_api/disaster/v2/views/object_class/spending.py

```diff
@@ -0,0 +1,139 @@
+from django.db.models import Q, Sum, Count, F, Value, Case, When
+from django.db.models.functions import Coalesce
+from rest_framework.response import Response
+
+from usaspending_api.awards.models import FinancialAccountsByAwards
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.data_classes import Pagination
+from usaspending_api.common.helpers.generic_helper import get_pagination_metadata
+from usaspending_api.disaster.v2.views.object_class.object_class_result import (
+    ObjectClassResults,
+    MajorClass,
+    MinorClass,
+)
+from usaspending_api.disaster.v2.views.disaster_base import (
+    DisasterBase,
+    PaginationMixin,
+    SpendingMixin,
+)
+from usaspending_api.financial_activities.models import FinancialAccountsByProgramActivityObjectClass
+
+
+def construct_response(results: list, pagination: Pagination):
+    object_classes = ObjectClassResults()
+    for row in results:
+        major_code = row.pop("major_code")
+        FA = MajorClass(id=major_code, code=major_code, description=row.pop("major_description"))
+        object_classes[FA].include(MinorClass(**row))
+
+    return {
+        "results": object_classes.finalize(pagination),
+        "page_metadata": get_pagination_metadata(len(object_classes), pagination.limit, pagination.page),
+    }
+
+
+class ObjectClassSpendingViewSet(PaginationMixin, SpendingMixin, DisasterBase):
+    """View to implement the API"""
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/disaster/object_class/spending.md"
+
+    @cache_response()
+    def post(self, request):
+        if self.spending_type == "award":
+            results = list(self.award_queryset)
+        else:
+            results = list(self.total_queryset)
+
+        return Response(construct_response(results, self.pagination))
+
+    @property
+    def total_queryset(self):
+        filters = [
+            Q(
+                Q(obligations_incurred_by_program_object_class_cpe__gt=0)
```

Comment:
time-permitting, this filter group can be replaced by `is_non_zero_object_class_cpe()` in CountBase